src/graph/UndirectedGraph.java

```diff
@@ -0,0 +1,27 @@
+package graph;
+
+public class UndirectedGraph extends Graph {
+
+  @Override
+  public Vertex removeNode(int id) {
+    Vertex v = super.removeNode(id);
+    if (v != null) {
+      for (int adjId : v.getNeighbors()) {
+        this.getNode(adjId).removeNeighbor(id);
```

Comment:
What if this.getNode(adjId) returns you a null if somehow the graph has not been contructed properly. Take care!!!

Reply:
Should I throw an exception in this case?

Reply:
You may. Anyway that acknowledges the problem.